webcompat/webhooks/__init__.py

```diff
@@ -33,9 +33,9 @@ def hooklistener():
     event_type = request.headers.get('X-GitHub-Event')
     # Treating events related to issues
     if event_type == 'issues':
-        issue_info = get_issue_info(payload)
+        webhook_issue = WebHookIssue(payload)
```

Comment:
Hm, that's a good question. It may have to do with the evolution of the design in my local commits.

Reply:
I will resolve this, since we now use `WebHookIssue.from_dict(payload)`.

Reply:
It's also possible to just do `WebHookIssue(action=payload.get('action'), body=issue_body, domain=domain, number=issue.get('number'), public_url=public_url, repository_url=issue.get('repository_url'), state=issue.get('state'), title=full_title, original_labels=original_labels, milestone=milestone, milestoned_with=milestoned_with)` (assuming the right variables are in scope).

But that offends *my* aesthetic sensibilities.

Reply:
yes that makes it a bit verbose :)

Reply:
> It's also possible to just do `WebHookIssue(action=payload.get('action'), body=issue_body, domain=domain, number=issue.get('number'), public_url=public_url, repository_url=issue.get('repository_url'), state=issue.get('state'), title=full_title, original_labels=original_labels, milestone=milestone, milestoned_with=milestoned_with)` (assuming the right variables are in scope).
> 
> But that offends _my_ aesthetic sensibilities.

I wouldn't recommend using this constructor in the codebase but it can be useful in tests. If you set default values for the fields in the dataclass declaration, this pattern can allow you to express the intention of a test more clearly and succinctly by only providing the fields necessary for a test, eg: `WebHookIssue(public_url='foo')`. (not that I'd recommend doing that in this case)

I think this pattern is generally called "alternate constructor" if you want to look it up.

Reply:
Yeah, that does seem nice to be able to be very explicit about intent in tests.